test/e2e/autoscale_test.go

```diff
@@ -357,12 +385,152 @@ func assertAutoscaleUpToNumPods(ctx *testContext, curPods, targetPods float64, d
 	}
 }
 
+func getUniqueHostPerConnection(t *testing.T, ctx *testContext, size int) (map[string]*websocket.Conn, error) {
+	clients := Setup(t)
+	uniqueHostConns := make(map[string]*websocket.Conn, size)
```

Comment:
this looks like a prime example of `sync.Map` usage ;)

Reply:
indeed lol ...

Reply:
came across the saga of this thread https://github.com/golang/go/issues/20680 on how `sync.Map` doesnt have a function to get the `length()` 😢

Reply:
But you don't need one, re: my comment above :)

Reply:
I still do in the test itself when doing the comparison. Unless you have another trick up your sleeve ;-)

Reply:
oh right since the size is predetermined now, I can get rid of the length checks in the test too. coolio